### PR TITLE
test(e2e): pre pull images

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
### Checklist prior to review

- [ ] [Link to relevant issue][1] as well as docs and UI issues --
- [ ] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: mk/e2e.new.mk [hunk 1]
```diff
@@ -23,6 +23,11 @@ E2E_UNIVERSAL_BIN_DEPS+= build/kumactl
 E2E_ENV_VARS+= GINKGO_EDITOR_INTEGRATION=true
 endif
 
+# List of images that we pull from the Internet in the E2E tests.
+# This list is a best effort, so we can pull and preload images to Kube clusters. In case of network error, we can fail fast.
+# In the future, we can cache those images so we won't pull this from external sources every single time.
+EXTERNAL_TEST_IMAGES = bitnami/kubectl:1.27.5 mendhak/http-https-echo:23 otel/opentelemetry-collector-contrib:0.92.0
```
Comment (lahabana) [inline mk/e2e.new.mk:29]: We should really get rid of: mendhak/http-https-echo I'm pretty sure our test image does this just fine

Comment (lahabana) [reply mk/e2e.new.mk:29]: https://github.com/kumahq/kuma/issues/9376

File: mk/e2e.new.mk [hunk 2]
```diff
@@ -70,6 +75,9 @@ test/e2e/k8s/start/cluster/$1:
 .PHONY: test/e2e/k8s/load/images/$1
 test/e2e/k8s/load/images/$1:
 	KIND_CLUSTER_NAME=$1 $(MAKE) $(K8S_CLUSTER_TOOL)/load/images
+ifdef CI
```
Comment (lahabana) [inline mk/e2e.new.mk:78]: Do we even need the CI? After all this will make little diff locally too no?

**Comment by jakubdyszkiewicz** [comment]
closing for now

**Pull request closed without merging**